### ISSUE TITLE
Type check errors

### ISSUE THREAD

**Issue by ashinkajay** [python-jsonschema/referencing#114]
Type check errors thrown by mypy when importing `referencing`

To reproduce this issue, create a `test.py` as below:
```
import referencing
```

Execute command:
```
mypy test.py
```

This issue is applicable on versions 0.31.0 and above.

**Comment by Julian** [comment]
```
⊙  echo 'import referencing' >foo.py && python3.12 -m venv venv && venv/bin/pip install --quiet referencing mypy && venv/bin/mypy foo.py                                                                                                                                                                            julian@Airm
Success: no issues found in 1 source file
```

**Comment by ashinkajay** [comment]
The type checks was done on a lower mypy version (0.961) That was the reason for the error.
